### ISSUE TITLE
Enhancement: [triple-slash-reference] support a new option named `never`

### ISSUE THREAD

**Issue by zanminkian** [typescript-eslint/typescript-eslint#11196]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/triple-slash-reference/

### Description

Current option is an object. I would like to see this rule accept a string option - `never`.

`nevert` option will totally disallow triple slash, including `/// <amd-module />` and `/// <reference no-default-lib="true"/>`

### Fail

```TypeScript
/// <amd-module />
/// <reference no-default-lib="true"/>
```

### Pass

```TypeScript
import foo from 'foo';
```
